### PR TITLE
Test global SLAM.

### PR DESCRIPTION
**Pull request by gaschler**

### CODE REVIEW

**Comment by gaschler** [comment]
On the current master, this test is flaky because RunFinalOptimization does not actually finish all computations, and the following check (sometimes) fails during deletion: [pose_graph.cc:51] Check failed: work_queue_ == nullptr
I believe the public API should allow to finish all computations and delete everything.
At first glance, #729 seems to fix this (and change the meaning of RunFinalOptimization to finish the entire work queue).

**Comment by cschuet** [comment]
As discussed let's fix the issue with the work queue first.

**Comment by gaschler** [comment]
@cschuet This can now be reviewed, the dependency is merged.

File: cartographer/mapping/map_builder_test.cc [hunk 1]
```diff
@@ -33,9 +33,9 @@ namespace {
 
 constexpr char kRangeSensorId[] = "range";
 constexpr char kIMUSensorId[] = "imu";
-constexpr double kDuration = 2.;
+constexpr double kDuration = 4.;
```
Comment (cschuet) [inline cartographer/mapping/map_builder_test.cc]: Should we add units to these constants?

**Comment by gaschler** [comment]
@wally-the-cartographer merge

**Comment by gaschler** [comment]
@wally-the-cartographer merge

**Pull request merged**